src/execution/operator/join/physical_join.cpp

```diff
@@ -17,6 +17,7 @@ bool PhysicalJoin::EmptyResultIfRHSIsEmpty() const {
 	case JoinType::INNER:
 	case JoinType::RIGHT:
 	case JoinType::SEMI:
+	case JoinType::RIGHT_ANTI:
```

Comment:
For some reason I remember taking this out because some tests with "IS NOT DISTINCT " conditions were failing. But those same issues would happen for right joins. so I'll add it back in